clang/test/dpct/helper_files_ref/include/device.hpp

```diff
@@ -435,37 +448,54 @@ Use 64 bits as memory_bus_width default value."
     queue = nullptr;
   }
   void set_saved_queue(sycl::queue* q) {
-    std::lock_guard<std::recursive_mutex> lock(m_mutex);
+    std::lock_guard<mutex_type> lock(m_mutex);
     _saved_queue = q;
   }
-  sycl::queue* get_saved_queue() const {
-    std::lock_guard<std::recursive_mutex> lock(m_mutex);
+  sycl::queue *get_saved_queue() const {
+    std::lock_guard<mutex_type> lock(m_mutex);
     return _saved_queue;
   }
   sycl::context get_context() const { return _ctx; }
 
 private:
-  sycl::property_list get_default_property_list_for_queue() const {
+  void clear_queues() {
+    _queues.clear();
+    _q_in_order = _q_out_of_order = _saved_queue = nullptr;
+  }
+
+  void init_queues() {
+    _q_in_order = create_in_order_queue_impl(true);
+    _q_out_of_order = create_out_of_order_queue_impl(true);
+    _saved_queue = &default_queue();
+  }
+
+  sycl::queue *create_in_order_queue_impl(bool enable_exception_handler) {
+    return create_queue_impl(enable_exception_handler,
```

Comment:
To make it simpler, let's remove create_in_order_queue_impl() and create_out_of_order_queue_impl(),  call create_queue_impl() directly.